File: backend/src/main/java/com/pikel/balancetracker/balance/model/Transaction.java
package com.pikel.balancetracker.balance.model;

import java.time.LocalDate;

public record Transaction(
        String sourceName,
        double amount,
        LocalDate date, // user inputs soonest date of payment
        TransactionType type, // debt or income
        PayPeriod payPeriod
) {
    public Transaction withDate(LocalDate newDate) {
        return new Transaction(sourceName, amount, newDate, type, payPeriod);
    }
}

File: frontend/types/transaction.ts

export type Transaction = {
    sourceName: string;
    amount: number;
    date: string;
    type: TransactionType; 
    payPeriod: PayPeriod;
};

export type TransactionEntity = Transaction &  {
    id: string;
};

export enum TransactionType {
    INCOME = "INCOME",
    DEBT = "DEBT"
};

export enum PayPeriod {
    WEEKLY = "WEEKLY",
    BIWEEKLY = "BIWEEKLY",
    MONTHLY = "MONTHLY",
    YEARLY = "YEARLY",
}